### PR TITLE
feat: add partner id integration

### PR DESCRIPTION
**Pull request by FiboApe**

### CODE REVIEW

File: src/core/frameworks/yearnSdk/index.ts
Comment (xgambitox) [inline src/core/frameworks/yearnSdk/index.ts:19]: Can we set 'partnerId' to 'undefined' better?

**Review by xgambitox — DISMISSED** [review]

**Review by turtlemoji — DISMISSED** [review]

**Comment by FoxTheSin** [comment]
amplify not working?
site isn't loading https://pr-514.d27dgpz01hmbvx.amplifyapp.com/

File: src/core/store/modules/tokens/tokens.reducer.ts
Comment (xgambitox) [inline src/core/store/modules/tokens/tokens.reducer.ts:174]: should we move all this logic to the vaults reducer instead?

File: src/core/store/modules/tokens/tokens.reducer.ts
Comment (xgambitox) [inline src/core/store/modules/tokens/tokens.reducer.ts:194]: should we move all this logic to the vaults reducer instead?

File: src/core/types/Service.ts
Comment (xgambitox) [inline src/core/types/Service.ts:187]: We should try to use a flat object when possible and not nested objects.

File: src/client/components/app/Transactions/DepositTx.tsx
Comment (xgambitox) [inline src/client/components/app/Transactions/DepositTx.tsx:75]:
```suggestion
  const [tokenAllowance, isLoadingTokenAllowance, tokenAllowanceError] = useAllowance(
```
Lets remove the `get` since it makes it look like if its a function

File: src/client/components/app/Transactions/Labs/LabDepositTx.tsx
Comment (xgambitox) [inline src/client/components/app/Transactions/Labs/LabDepositTx.tsx:71]:
```suggestion
  const [tokenAllowance, isLoadingTokenAllowance, tokenAllowanceError] = useAllowance(
```
Lets remove the `get` since it makes it look like if its a function

File: src/client/hooks/useAllowance.ts
Comment (xgambitox) [inline src/client/hooks/useAllowance.ts:13]: We should try to use a flat object when possible and not nested objects.

File: src/client/context/AppServiceContext.tsx
Comment (xgambitox) [inline src/client/context/AppServiceContext.tsx:13]: To maintain the currents system architecture, we should avoid UI calling services directly, we use redux actions as an adapter layer. As an alternative, we can dispatch a thunk and we can unwrap the value, with no reducer needed.

File: src/client/hooks/useAllowance.ts
Comment (xgambitox) [inline src/client/hooks/useAllowance.ts:35]: remove console log

File: src/client/context/AppServiceContext.tsx
Comment (xgambitox) [reply src/client/context/AppServiceContext.tsx:13]: Since we are out of time for this feature, we can leave it for now and aim to refactor it immediately after release.

File: src/core/frameworks/yearnSdk/index.ts
Comment (xgambitox) [inline src/core/frameworks/yearnSdk/index.ts:22]: can remove

File: src/core/types/Service.ts
Comment (FiboApe) [reply src/core/types/Service.ts:187]: sure will change that

File: src/client/components/app/Transactions/DepositTx.tsx
Comment (FiboApe) [reply src/client/components/app/Transactions/DepositTx.tsx:75]: will do

File: src/client/context/AppServiceContext.tsx
Comment (FiboApe) [reply src/client/context/AppServiceContext.tsx:13]: what's the logic behind not allowing hooks or ui not call services directly? It seems that just adding a redux action as an intermediary is overhead just to keep a pattern that might not make sense in all cases.

It seems like we would just use redux for the thunk action but actually not do anything with the result from it

File: src/client/components/app/Transactions/Labs/LabDepositTx.tsx
Comment (FiboApe) [reply src/client/components/app/Transactions/Labs/LabDepositTx.tsx]: solved on 6ad01db108237880b69ecb49695a74b419be2063

File: src/client/components/app/Transactions/DepositTx.tsx
Comment (FiboApe) [reply src/client/components/app/Transactions/DepositTx.tsx]: solved on 6ad01db108237880b69ecb49695a74b419be2063

File: src/client/hooks/useAllowance.ts
Comment (FiboApe) [reply src/client/hooks/useAllowance.ts]: solved on fff3e86f6179d4e466b091c8f555551d66f809ef

File: src/core/types/Service.ts
Comment (FiboApe) [reply src/core/types/Service.ts]: solved on fff3e86f6179d4e466b091c8f555551d66f809ef

File: src/client/hooks/useAllowance.ts
Comment (FiboApe) [reply src/client/hooks/useAllowance.ts]: solved on 305142a804e3c9fe05a74cb42aecce53b2d45f4e

File: src/core/frameworks/yearnSdk/index.ts
Comment (FiboApe) [reply src/core/frameworks/yearnSdk/index.ts]: solved on ecc064b992e17714c48485b4293dee5e8c5f2590

File: src/client/context/AppServiceContext.tsx
Comment (xgambitox) [reply src/client/context/AppServiceContext.tsx:13]: We are applying the [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html) principles. We are trying to maintain the business logic independent of the UI. Maintaining these separation between layers, each having their own responsibilities, make code more maintainable, flexible and scalable. In this case, actions are part of our Interface Adapters layer.

If you take a look at how it was implemented before this change, it was following this pattern. Although maintaining this architecture can be seen as having too much boilerplate, we designed it this way so that we get all the benefits from it, including utilities from the tooling itself. We add validations there, since we dont want to only have them on UI side, also makes them callable by other thunks and not just directly from UI components. Right now we are missing some critical stuff that will also benefit from this structure, like error handling and analytics through middleware, and of course testing for the core logic (actions, reducers, services).

File: src/core/store/modules/vaults/vaults.actions.ts
Comment (xgambitox) [inline src/core/store/modules/vaults/vaults.actions.ts:204]: Why these changes? Is it even called now that the service is called directly?

File: src/core/store/modules/vaults/vaults.actions.ts
Comment (xgambitox) [inline src/core/store/modules/vaults/vaults.actions.ts:233]: same comment [here](https://github.com/yearn/yearn-finance-v3/pull/514/files#r838029524)  for this changes

File: src/client/hooks/useAllowance.ts
Comment (karelianpie) [inline src/client/hooks/useAllowance.ts:12]:
```suggestion
  tokenAddress?: string,
```

File: src/client/hooks/useAllowance.ts
Comment (karelianpie) [inline src/client/hooks/useAllowance.ts:11]: Can we have here named params?

Comment (FiboApe) [reply src/client/hooks/useAllowance.ts]: solved on ffefab04728c259b147ce88165c34d609bd1c7b4

File: src/client/context/AppServiceContext.tsx
Comment (FiboApe) [reply src/client/context/AppServiceContext.tsx]: solved on ffefab04728c259b147ce88165c34d609bd1c7b4

File: src/core/store/modules/vaults/vaults.actions.ts
Comment (gotzenx) [inline src/core/store/modules/vaults/vaults.actions.ts:192]: should this be `!==`?

**Review by gotzenx — DISMISSED** [review]

File: src/core/store/modules/vaults/vaults.actions.ts
Comment (FiboApe) [reply src/core/store/modules/vaults/vaults.actions.ts]: fixed with 020ad2bba3da6fa711b571e9e3cbee69c0e3b331

File: src/core/types/Service.ts
Comment (xgambitox) [inline src/core/types/Service.ts:60]: With new refactor on sdk, we can remove the `Boolean` from return type
```suggestion
  approveDeposit: (props: ApproveDepositProps) => Promise<TransactionResponse>;
  approveZapOut: (props: ApproveZapOutProps) => Promise<TransactionResponse>;
```

File: src/core/services/VaultService.ts
Comment (xgambitox) [inline src/core/services/VaultService.ts:275]: the Token argument should be the intended target token to withdraw to. Its not always the underlying token (vault.token). Would suggest we just receive directly the addresses instead of the whole Vault data structure.

File: src/core/types/Service.ts
Comment (karelianpie) [reply src/core/types/Service.ts:60]: Also, we should always use lowercase `boolean` as that's the primitive type (not `Boolean`).

File: src/core/services/VaultService.ts
Comment (karelianpie) [inline src/core/services/VaultService.ts:264]:
```suggestion
  public async approveDeposit(props: ApproveDepositProps): Promise<boolean | TransactionResponse> {
```

File: src/core/types/Service.ts
Comment (xgambitox) [reply src/core/types/Service.ts:60]: Yes, that was an error on the SDK that was removed in the recent version, so Fibo probably was forced to copy that interface. Agree we should stick with primitive. Also we should not have this type of interfaces that can return multiple types.

File: src/core/services/VaultService.ts
Comment (FiboApe) [reply src/core/services/VaultService.ts]: fixed on 227dce8ccc7427637245d1ffbe9f21f0819cece8

File: src/core/services/VaultService.ts
Comment (FiboApe) [reply src/core/services/VaultService.ts]: fixed on 227dce8ccc7427637245d1ffbe9f21f0819cece8

File: src/core/types/Service.ts
Comment (FiboApe) [reply src/core/types/Service.ts]: fixed on 227dce8ccc7427637245d1ffbe9f21f0819cece8

File: src/core/store/modules/vaults/vaults.actions.ts
Comment (xgambitox) [inline src/core/store/modules/vaults/vaults.actions.ts:203]: Can remove this now that TX type changed

File: src/core/store/modules/vaults/vaults.actions.ts
Comment (xgambitox) [inline src/core/store/modules/vaults/vaults.actions.ts:207]: Since spender can be also partner tracking contract, would be best to dispatch `getVaultAllowance` after transaction is approved, so that state gets updated correctly ?

Comment (FiboApe) [reply src/core/store/modules/vaults/vaults.actions.ts]: fixed on b69e516c61c60f31501bd291946417d970f1b2ad

File: src/core/store/modules/vaults/vaults.actions.ts
Comment (FiboApe) [reply src/core/store/modules/vaults/vaults.actions.ts]: fixed on b69e516c61c60f31501bd291946417d970f1b2ad

**Review by xgambitox — APPROVED** [review]

**Comment by FoxTheSin** [comment]
updated testing may have come across new error with approval
image_2022-04-07_15-21-26
image_2022-04-07_15-17-58

**Review by xgambitox — APPROVED** [review]

**Comment by FoxTheSin** [comment]
errors on approve appear resolved. 
tested fantom deposit look good
tested eth deposit simulations look good

**Review by FoxTheSin — APPROVED** [review]

**Pull request merged**
